rust/loqui_client/src/connection_handler.rs

```diff
@@ -140,13 +139,19 @@ impl<E: Encoder> Handler for ConnectionHandler<E> {
         let TransportOptions { encoding, .. } = transport_options;
         // Forward Request and Push events to the connection so it can send them to the server.
         match event {
-            InternalEvent::Request { payload, waiter_tx } => {
+            InternalEvent::Request { payload, waiter } => {
                 let sequence_id = id_sequence.next();
-                self.handle_request(payload, encoding, sequence_id, waiter_tx)
+                self.handle_request(payload, encoding, sequence_id, waiter)
             }
             InternalEvent::Push { payload } => self.handle_push(payload, encoding),
         }
     }
+
+    fn handle_ping(&mut self) {
```

Comment:
Maybe we should call this like `handle_ping_sent`? or something - to let you know that "at this point, the ping was sent."

Reply:
This is actually on receive, so I think it's okay to keep it `handle_ping`.